python/cudf/cudf/_lib/parquet.pyx

```diff
@@ -698,7 +698,14 @@ cdef _set_col_metadata(
     column_in_metadata& col_meta,
     bool force_nullable_schema
 ):
-    col_meta.set_nullability(force_nullable_schema or col.nullable)
+    if force_nullable_schema or not col.nullable:
```

Comment:
I’m not sure if I agree with the comment below. Here is what I expected:
1. Col is nullable and forced nullable. This will be written as nullable by default so no action is needed.
2. Col is nullable and not forced nullable. This will be written as nullable by default so no action is needed.
3. Col is not nullable and forced nullable. The call to tell C++ to write nulls is needed.
4. Col is not nullable and not forced nullable. No action is needed.

So can we write this?
```python
if force_nullable_schema and not col.nullable:
    col_meta.set_nullability(True)
```

Reply:
All are right, except this:

> 4. Col is not nullable and not forced nullable. No action is needed.

We need action for this incase of chunked parquet writer. Because if `_nullability` isn't defined, the default libcudf behavior is to return `null` schema. So when someone asks for `force_nullable_schema=False` & column isn't having any nulls the current logic will yield `not null` as requested by the user.

Reply:
Okay. The conditional can still be simplified if only cases 3 and 4 (with non-nullable columns) need action:
```suggestion
    if not col.nullable:
```

Reply:
Logically I feel this is right, but seems like there is something else going on too..investigating..

Reply:
After doing some investigation @vuule found a bug on libcudf parquet writer side for a sliced struct column. We decided to scope down the `force_nullable_schema` to only single writer instead. Thus, dropping it's support in chunked parquet writer will not let the bug surface. We don't want to introduce wide-ranging changes at parquet writer side to address the actual problem. So this PR should now be ready for review.